src/WinRT.Runtime/TypeNameSupport.cs

```diff
@@ -25,12 +26,34 @@ enum TypeNameGenerationFlags
     static class TypeNameSupport
     {
         private static List<Assembly> projectionAssemblies = new List<Assembly>();
+        private static ConcurrentDictionary<string, Type> typeNameCache = new ConcurrentDictionary<string, Type>() { ["TrackerCollection<T>"] = null };
```

Comment:
Any concern this might keep all encountered types alive and prevent it from being unloaded?

Reply:
I thought we'd discuss that last year, and decided that caching types wasn't a concern

Reply:
Right I think we discussed that when we starting doing something similar with TypeObjectRefFieldCache / TypeObjectRefFuncCache.  We eventually did stop doing that for the net5 projection but that was because we were able to do further optimizations that made it not need a cache and not due to type unloadability concerns.  So this might be fine.  I believe we said then types are rarely unloaded or something.